index.html

```diff
@@ -831,6 +835,24 @@ <h4><dfn>SmartCardConnectOptions</dfn> enum</h5>
       <section>
         <h3>Auxiliary algorithms and definitions</h4>
 
+        <p>To <dfn>clear the operationInProgress</dfn> of a
+        {{SmartCardContext}} |context:SmartCardContext|, perform the following steps:</p>
+        <ol>
+          <li>[=Assert=]: |context|.{{SmartCardContext/[[operationInProgress]]}}
+            is `true`.</li>
+          <li>Set |context|.{{SmartCardContext/[[operationInProgress]]}} to
+            `false`.</li>
+          <li>[=set/For each=] |connection:SmartCardConnection| of
+            |context|.{{SmartCardContext/[[connections]]}}:
+            <ol>
+              <li>[=SmartCardConnection/End any pending transaction=] of
```

Comment:
I think this is correct but requires some comments to explain the motivation here. The important detail about the "end any pending transaction" algorithm is that it only ends a transaction where the promise returned by the transaction callback has settled and therefore has a disposition. Such transactions may be waiting for the context to be idle so that `SCardEndTransaction` can be called. Maybe renaming this algorithm to "end any settled transaction" to make that distinction clearer. A non-normative note either on this algorithm or on the transaction state struct explaining the state transitions a transaction can go through would potentially also help readers.

Reply:
Oh, I see the confusion now. It's worded as if the transaction ifself was pending. But actually what is pending is the operation to **end** that transaction. That new name for the algorithm is much better indeed.

And I agree that an explanation note is warranted there.

Done.